### PR TITLE
[Microsoft.Billing] Add tenantId property to product.

### PR DESCRIPTION
**Pull request by wilcobmsft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 1]
```diff
@@ -6546,6 +6546,11 @@
           "type": "string",
           "readOnly": true
         },
+        "tenantId": {
+          "description": "The id of the tenant in which the product is used.",
```
Comment (asarkar84) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: Shall tenant id be populated in all kind of products or just AAD premium. There will be PowerBI and Office in future.
Would it be good to specify that the tenant id will be supported/specified for AAD premium only?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ProductsListByBillingAccount.json [hunk 2]
```diff
@@ -25,6 +25,8 @@
               "quantity": 4,
               "skuId": "0001",
               "skuDescription": "Enterprise Agreement Development",
+              "tenantId": "515a6d36-aaf8-4ca2-a5e8-c45deb0c5cce"
+              "tenantId": "515a6d36-aaf8-4ca2-a5e8-c45deb0c5cce"
```
Comment (asarkar84) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ProductsListByBillingAccount.json]: Added twice

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ProductsListByBillingAccount.json [hunk 3]
```diff
@@ -52,6 +54,8 @@
...
```
Comment (asarkar84) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ProductsListByBillingAccount.json]: Added twice

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json (hunk 1 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: This should work for any product with a tenant association.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ProductsListByBillingAccount.json (hunk 3 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ProductsListByBillingAccount.json]: Fixed.

**Review by asarkar84 — APPROVED** [review]

**Review by allenjzhang — APPROVED** [review]

**Pull request merged**